runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java

```diff
@@ -274,21 +274,31 @@ else if (action.getSprite() != null)
 				@Override
 				public void mousePressed(MouseEvent e)
 				{
-					if (!e.isShiftDown())
+					//remove all slots when a selection is chosen without shift or control
+					if (!e.isShiftDown() && !e.isControlDown())
 					{
 						clearCombinedSlots();
 					}
 
-					if (slot.isSelected())
+					//If control is pressed, want to add, even if it has already been added
+					if (slot.isSelected() && !e.isControlDown())
 					{
 						combinedActionSlots.remove(slot);
+						slot.setSelected(false);
 					}
 					else
 					{
 						combinedActionSlots.add(slot);
 					}
 
-					slot.setSelected(!slot.isSelected());
+					// Loop through the array and set the slot to selected if any element matches
+					for (UIActionSlot list : combinedActionSlots)
```

Comment:
This is just `slot.setSelected(true);`

Reply:
The confusion you had lead to me changing the method to slot.setSelected(combinedActionSlots.contains(slot)), which is a cleaner way to accomplish what I desired.

Reply:
You can just put it in the if/else and set it to true/false on each side.

Reply:
I did try that, but ran into a problem when testing it. The arrayList can contain several entries of the item (slot), so I can't just set selected to false when I remove one item from the list, cause then it won't let me remove others. Based on the code structure, the selected flag has to mean exists in the list, which when there are multiple entries essentially is list.contains

Reply:
Ah